### DISCUSSION TITLE
Microsoft teams not being assigned to a window.

### DISCUSSION THREAD

**Discussion by haddadnidal** [i3/i3#4788] [Q&A] [answered]
When launching microsoft teams and it's being assigned to a window using ` assign [class="microsoft"] 5 ` is still shown on the current window. 
I used `xprop | grep 'CLASS'` and the result was `WM_CLASS(STRING) = "microsoft teams - preview", "Microsoft Teams - Preview"`

i tried  ` assign [class="microsoft teams - preview"] 5 ` and i had the same behavior. 

I dont know if this is usefull but `xprop`on the popup launch window result in this : 
```
_NET_WM_DESKTOP(CARDINAL) = 1
_NET_WM_STATE(ATOM) = _NET_WM_STATE_FOCUSED
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
I3_FLOATING_WINDOW(CARDINAL) = 1
_NET_WM_USER_TIME(CARDINAL) = 3365451
WM_NAME(UTF8_STRING) = "Loading Microsoft Teams"
_NET_WM_NAME(UTF8_STRING) = "Loading Microsoft Teams"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 2106, 416
		program specified minimum size: 440 by 248
		program specified maximum size: 440 by 248
_NET_WM_ICON(CARDINAL) = 
XdndAware(ATOM) = BITMAP
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x0, 0x0, 0x0
_NET_WM_BYPASS_COMPOSITOR(CARDINAL) = 2
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 31457289, 31457290
_GTK_THEME_VARIANT(UTF8_STRING) = "dark"
_GTK_HIDE_TITLEBAR_WHEN_MAXIMIZED(CARDINAL) = 1
WM_WINDOW_ROLE(STRING) = "browser-window"
WM_CLASS(STRING) = "microsoft teams - preview", "Microsoft Teams - Preview"
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_PID(CARDINAL) = 10621
WM_CLIENT_MACHINE(STRING) = "debian"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING, _NET_WM_SYNC_REQUEST
```
Thank you very much.

**Comment by Airblader** [comment, marked as answer]
Some clients misbehave and change their `WM_CLASS` after mapping the window, which they aren't allowed to do (Spotify is known to do this, too). A workaround, if that is the case, would be to use

```
for_window [class=microsoft] move window to workspace 5
```

instead. An i3 log file would be needed to confirm this is what is going on.

**Reply by orestisfl** [reply]
The first class item is the instance, right? So it should be Microsoft not microsoft

**Reply by haddadnidal** [reply]
Thank you @Airblader for the solution i modified it and put `"Microsoft"`
